net/net.go

```diff
@@ -218,6 +218,7 @@ func (n *net) CreateThread(_ context.Context, id thread.ID, opts ...core.NewThre
 	if !info.Key.Defined() {
 		info.Key = thread.NewRandomKey()
 	}
+	// This may overwrite ThreadKey information if there is a mismatch
```

Comment:
Note that some existing tests relied on the ability to overwrite read and service keys, so no check or error is thrown when this happens. This _seems_ ok, because it would support things like key rotation etc... but I've added this comment as a reminder and perhaps this should be documented in the public APIs?

Reply:
What about only adding new keys if they don't already exists? So, modifying `logstore.AddThread` to not blindly overwrite?

Reply:
So silently ignore if new/mismatched keys are used? That's an option for sure, would it lead to unexpected results?

Reply:
Maybe just error if something would be overwritten?

Reply:
Yeh that's what I had, but we have some existing tests that rely on being able to overwrite. Hence my comment above... unless I'm missing something here?

Reply:
In particular, this test: https://github.com/textileio/go-threads/blob/master/db/db_test.go#L94

Reply:
We could modify `NewDB` to allow passing in ThreadOptions, so that you could specify the same keys again, and if you didn't, return an error.

Reply:
That's what I went with.